step-templates/upgrade-octopus-server.json

```diff
@@ -3,16 +3,13 @@
-    "Octopus.Action.Script.ScriptFileName": null,
-    "Octopus.Action.Package.FeedId": null,
-    "Octopus.Action.Package.PackageId": null
+    "Octopus.Action.Script.ScriptBody": "[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12\n\nFunction GetLatestVersion(){\n    if([string]::IsNullOrEmpty($SpecificVersionInstallation)){\n        Write-Host \"SpecificVersionInstallation is empty, getting latest version from octopusdeploy.com\"\n        $versions = Invoke-WebRequest https://octopusdeploy.com/download/upgrade/v3 -UseBasicParsing | ConvertFrom-Json\n        $version = $versions[-1].Version\n        return $version\n    }\n    else {\n        Write-Host \"Specifc version is selected, chosen version \" $SpecificVersionInstallation\n        return $SpecificVersionInstallation\n    }\n}\n\nFunction GetInstalledVersion () {\n    $InstalledVersion = (get-item \"$InstallPath\\Octopus.Server.exe\").VersionInfo.fileversion\n    return $installedVersion\n}\n\nFunction DownloadOctopusDeploy([string] $versionNumber) {\n   \n    Write-Host \"Downloading Octopus $version\"\n    $tempFile = [System.IO.Path]::GetTempFileName()\n  \ttry\n    {\n        (New-Object System.Net.WebClient).DownloadFile(\"https://download.octopusdeploy.com/octopus/Octopus.$version-x64.msi\", $tempFile)\n    }\n    catch\n    {\n        Write-Host \"Exception occured\"\n        echo $_.Exception|format-list -force\n    }\n\n    Write-Host \"Download completed\"\n    return $tempFile\n}\n\n\nFunction StopOctopusServer() {\n    Write-Host \"Stopping Server\"\n    . \"$InstallPath\\Octopus.Server.exe\" service --stop --console --instance $InstanceName\n}\n\nFunction InstallOctopusDeploy([object] $tempFile){\n    Write-Host \"Installing\"\n    msiexec /i $tempFile /quiet | Out-Null\n}\n\nFunction RemoveTempFile([object] $tempFile)\n{   \n    Write-Host \"Deleting installer\"\n    Remove-Item $tempFile\n}\n\nFunction StartOctopusDeploy(){\n    Write-Host \"Starting Server\"\n    . \"$InstallPath\\Octopus.Server.exe\" service --start --console --instance $InstanceName\n}\n\n\n$version = GetLatestVersion\n$installedVersion = GetInstalledVersion\n\n\nif([version]$version -eq [version]$installedVersion){\n    Write-host \"Octopus already updated to version: $version\"   \n}\n\nelse  {\n    Write-Host \"Installed version is $installedVersion\"\n    Write-Host \"Installing version $version\" \n\n    $tempFile = DownloadOctopusDeploy $version\n    StopOctopusServer\n    InstallOctopusDeploy $tempFile\n    RemoveTempFile $tempFile\n    StartOctopusDeploy\n}\n\n"
```

Comment:
With these three functions `DownloadOctopusDeploy`, `StopOctopusServer` and `StartOctopusDeploy`, I think it would be good if they all followed the same naming scheme `<Action>OctopusServer`, so it'd be `DownloadOctopusServer`, `StopOctopusServer` and `StartOctopusServer`.